packages/block-library/src/navigation/view.js

```diff
@@ -62,18 +62,20 @@ const { state, actions } = store(
 					// Only open on hover if the overlay is closed.
 					Object.values( overlayOpenedBy || {} ).filter( Boolean )
 						.length === 0
-				)
+				) {
 					actions.openMenu( 'hover' );
+				}
 			},
 			closeMenuOnHover() {
 				const { type, overlayOpenedBy } = getContext();
 				if (
 					type === 'submenu' &&
-					// Only open on hover if the overlay is closed.
+					// Only close on hover if the overlay is closed.
```

Comment:
I'm confused as to why we close on hover when something is closed...

Reply:
Because if overlay is open the submenu should not be closed as it's always expanded.